patches/tModLoader/Terraria/ModLoader/TMLLootDatabase.cs

```diff
@@ -0,0 +1,831 @@
+using System.Collections.Generic;
+using System.Linq;
+using Terraria.GameContent;
+using Terraria.ID;
+using Terraria.Localization;
+
+namespace Terraria.ModLoader
+{
+	// can and/or will be used for other tml loots later. (ie pots loot)
+	public class TMLLootDatabase
+	{
+		private Dictionary<int, ChestLoot> npcShopByType = new();
+		private List<ChestLoot.Entry> globalNpcShopEntries = new();
```

Comment:
proposed is `private Dictionary<string, ChestLoot> npcShopByName = new();`
and when registering for a mod, we need `ModNPC.FullName + "/" + ShopTitle`
For vanilla, let's use Terraria/{NPCName}/ShopTitle

By default, when registering a new Shop, shop title is 'Shop'. Modder can change during setup shop step.